### ISSUE TITLE
Upgrade to SQLite 3.43 in macOS and Windows installers

### ISSUE THREAD

**Issue by erlend-aasland** [python/cpython#109286]
# Feature or enhancement

3.43.0 was released a couple of weeks ago; 3.43.1 was released recently. There's no rush; let's wait a couple of weeks to see if more patch releases appear before creating PRs.

https://sqlite.org/releaselog/3_43_1.html

### Linked PRs
* gh-110252
* gh-110374
* gh-110399
* gh-110403
* gh-110478
* gh-110479
* gh-110482
* gh-110550
* gh-110551

**Comment by Harshchitaliya** [comment]
sudo pip install --upgrade pysqlite

**Comment by erlend-aasland** [comment]
FTR, there haven't been further updates on 3.43 recently, so maybe we should consider upgrading now.

**Comment by jtranquilli** [comment]
Could I be assigned this issue?

**Comment by erlend-aasland** [comment]
> Could I be assigned this issue?

Sure, go ahead! You might want to take a look at the previous SQLite version bumps, and also [the devguide](https://devguide.python.org):

- gh-104623
- gh-102997
- gh-101759

**Comment by erlend-aasland** [comment]
FTR, I've bumped the sources repo to SQLite 3.43.1, meaning the Windows CI is now ready for the update.

**Comment by jtranquilli** [comment]
Working on this now, plan is to update 4 files.
1.  Update line 57 of PCbuild/get_externals.bat to reflect the change in SQlite versions 
2. Update line 362 of Mac/BuildScript/build-installer.py to reflect the change in SQlite versions
3. Update line 363 of Mac/BuildScript/build-installer.py to change the download URL to the URL of the new version
4. Update line 1 of Misc/NEWS.d/next/macOS/2023-05-18-22-31-49.gh-issue-104623.6h7Xfx.rst to reflect the change in SQlite versions

I noticed that in some but not all of the previous commit requests from SQlite upgrades, a variable called checksum (line 364, build-installer.py) was updated. I wasn't able to figure out what this variable is representing or whether it needs to be changed, so for the time being I will leave it as is.
